tests/hifimags.py

```diff
@@ -0,0 +1,374 @@
+    {"PV": "READY", "EXTRA_READ_PV":"READY:RBV", "values":READYS, "init_value":""},
+    {"PV": "OUTPUT:FIELD:GAUSS", "EXTRA_READ_PV": "OUT:RBV", "values":GAUSS, "init_value":-4},
+    {"PV": "OUTPUT:CURR", "EXTRA_READ_PV": "", "values":AMPSANDVOLTS, "init_value":-0.4},
+    {"PV": "OUTPUT:VOLT", "EXTRA_READ_PV": "", "values":AMPSANDVOLTS, "init_value": -0.4},
+    {"PV": "TARGET:TIME", "EXTRA_READ_PV": "", "values":TIMES, "init_value":""},
+    {"PV": "QUENCH", "EXTRA_READ_PV": "", "values":QUENCHES, "init_value":""},
+    #{"PV": "", "EXTRA_READ_PV": "", "values": "", "init_value":""},
+]
+WRITE_PVS = [
+    {"MAG": "X", "PV": "TARGET", "EXTRA_READ_PV": "", "values": GAUSS, "init_value": -4},
+    {"MAG": "M", "PV": "RAMP:LEADS", "EXTRA_READ_PV": "", "values": LEADS, "init_value": ""},
+    {"MAG": "M", "PV": "PERSIST", "EXTRA_READ_PV": "", "values": PERSISTS, "init_value": ""},
+]
+WRITE_SWITCH_PVS = [
+    {"PV": "RAMP:RATE", "EXTRA_READ_PV": "", "values": SWITCHINGRAMPRATE, "init_value": ""},
+    {"PV": "FIELD:MAX", "EXTRA_READ_PV": "", "values": SWITCHINGMAX, "init_value": ""},
+    {"PV": "FIELD:MID", "EXTRA_READ_PV": "", "values": SWTICHINGMID, "init_value": ""},
+]
+MAIN_PVS = [
+    {"PV": "STAT", "EXTRA_READ_PV":"MAIN:STAT:RBV", "values":STATUSES, "init_value": ""},
+    {"PV": "READY", "EXTRA_READ_PV": "MAIN:READY:RBV", "values": READYS, "init_value": ""},
+    {"PV": "OUTPUT:FIELD:GAUSS", "EXTRA_READ_PV": "MAIN:OUT:RBV", "values":GAUSS, "init_value":-4},
+    {"PV": "TARGET:TIME", "EXTRA_READ_PV": "MAIN:MINTO:RBV", "values":TIMES, "init_value":""},
+    {"PV": "OUTPUT:FIELD:PERSIST:GAUSS", "EXTRA_READ_PV": "", "values":GAUSS, "init_value":""},
+]
+SYS_READ_PVS = [
+    {"PV": "HALL:SENS1", "EXTRA_READ_PV": "HALL:SENS1:RBV", "values": HALLS, "init_value": -1},
+    {"PV": "HALL:SENS2", "EXTRA_READ_PV": "HALL:SENS2:RBV", "values": HALLS, "init_value": -1},
+    {"PV": "COMP:R:ERROR:TEXT:RBV", "EXTRA_READ_PV": "", "values": COMP_TEXTS, "init_value": "Testing"},
+    {"PV": "COMP:L:ERROR:TEXT:RBV", "EXTRA_READ_PV": "", "values": COMP_TEXTS, "init_value": "Testing"},
+    {"PV": "COMP:R:ERROR:STAT:RBV", "EXTRA_READ_PV": "", "values": ERRORS, "init_value": "Error"},
+    {"PV": "COMP:L:ERROR:STAT:RBV", "EXTRA_READ_PV": "", "values": ERRORS, "init_value": "Error"},
+    {"PV": "COMP:R:ON:RBV", "EXTRA_READ_PV": "", "values": ONOFF, "init_value": "On"},
+    {"PV": "COMP:L:ON:RBV", "EXTRA_READ_PV": "", "values": ONOFF, "init_value": "On"},
+]
+SYS_WRITE_PVS = [
+    {"PV": "OPMODE", "EXTRA_READ_PV": "OPMODE:RBV", "values": OPMODES, "init_value": "High Field"},
+]
+
+class HifimagsTests(unittest.TestCase):
+    """
+    Tests for the Hifimags IOC.
+    """
+    def setUp(self):
+        self.ca = ChannelAccess(device_prefix=DEVICE_PREFIX)
+
+    def setZeroes(self):
+        for PSU in PSUS:
+            self.ca.set_pv_value(PSU + ":ZERO", 0.2)
+            self.ca.set_pv_value(PSU + ":ZEROFIELD", 0.3)
```

Comment:
zeroes here are a value that might not actually be zero, I set them to a specific value for the tests so that I can be sure they are actually applied.